x-pack/plugins/lens/public/datasources/text_based/text_based_languages.tsx

```diff
@@ -233,13 +233,14 @@ export function getTextBasedDatasource({
       });
       return { state: { layers }, savedObjectReferences };
     },
-    isValidColumn(state, indexPatterns, layerId, columnId) {
-      const layer = state.layers[layerId];
-      const column = layer.columns.find((c) => c.columnId === columnId);
-      const indexPattern = indexPatterns[layer.index];
-      if (!column || !indexPattern) return false;
-      return true;
-    },
+    // TODO - make sure this is covered in the user messsages
+    // isValidColumn(state, indexPatterns, layerId, columnId) {
+    //   const layer = state.layers[layerId];
+    //   const column = layer.columns.find((c) => c.columnId === columnId);
+    //   const indexPattern = indexPatterns[layer.index];
+    //   if (!column || !indexPattern) return false;
+    //   return true;
+    // },
```

Comment:
@stratoula I'd like your input here.

This method performed a column existence check and an index pattern existence check. The result was used to mark the dimension button invalid.

To me, it looks like we may be able to delete this without replacement.

The column check is covered in the [`renderDimensionTrigger`](https://github.com/elastic/kibana/blob/da0b63b3e56a318e61f1a6ce36723bd4a07c9b76/x-pack/plugins/lens/public/datasources/text_based/text_based_languages.tsx#L388) method and I've verified that the UI looks the same.

The last thing I'm wondering: do you know why the index pattern check is in here? E.g. is there a way to hit this branch?

Reply:
Yes @drewdaemon we can clear this out